reporting/test_report.py

```diff
@@ -0,0 +1,188 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from __future__ import annotations
+
+import json
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from datetime import datetime
+from enum import Enum
+from typing import TYPE_CHECKING, ClassVar, List, Optional, Union
+
+import asyncer
+from ..utils.dagger import with_exit_code, with_stderr, with_stdout
+from ..models.contexts import GlobalContext
+from ..models.pipelines import StepResult
+from dagger import Client, Container, QueryError
+from rich import Console
+from rich.console import Group
+from rich.panel import Panel
+from rich.style import Style
+from rich.table import Table
+from rich.text import Text
+
+console = Console()
+
+
+@dataclass(frozen=True)
+class GlobalTestReport:
```

Comment:
```suggestion
class GlobalReport:
```
I think we should have two kind of reports: pipeline specific ones that will provide details about a pipeline execution, and a global one that would aggregate the reports of all the pipelines that got executed in the global context. I think `GlobalContext` and `GlobalReport` should be singletons. `GlobalContext` holds all the pipelines contexts that need to run, GlobalReport gather the execution results, shared through pipeline specific reports.